services/github/github-tag.service.js

```diff
@@ -84,39 +76,81 @@ module.exports = class GithubTag extends GithubAuthService {
     return { label: 'tag' }
   }
 
-  static render({ usingSemver, version }) {
+  static render({ version, sort }) {
     return {
       message: addv(version),
-      color: usingSemver ? versionColor(version) : 'blue',
-    }
-  }
-
-  static transform({ usingSemver, includePre, json }) {
-    const versions = json.map(({ name }) => name)
-    if (usingSemver) {
-      return latest(versions, { pre: includePre })
-    } else {
-      return versions[0]
+      color: sort === 'semver' ? versionColor(version) : 'blue',
     }
   }
 
-  async handle({ variant, user, repo }) {
-    const usingSemver = variant !== 'tag-date'
-    const includePre = variant === 'tag-pre'
-
-    const json = await this._requestJson({
+  async fetch({ user, repo }) {
+    return this._requestJson({
       url: `/repos/${user}/${repo}/tags`,
       schema,
-      errorMessages: errorMessagesFor(),
+      errorMessages: errorMessagesFor('repo not found'),
     })
-    if (json.length === 0) {
-      throw new NotFound({ prettyMessage: 'none' })
+  }
+
+  static getLatestTag({ tags, sort, includePrereleases }) {
+    if (sort === 'semver') {
+      return latest(tags.map(tag => tag.name), { pre: includePrereleases })
     }
-    const version = this.constructor.transform({
-      usingSemver,
-      includePre,
-      json,
+    return tags[0].name
+  }
+
+  async handle({ user, repo }, queryParams) {
+    const sort = queryParams.sort
+    const includePrereleases = queryParams.include_prereleases !== undefined
+
+    const tags = await this.fetch({ user, repo })
+    if (tags.length === 0) throw new NotFound({ prettyMessage: 'none' })
```

Comment:
Same suggestions as above in the `release` service to update the message and consider moving this check into `getLatestTag` for easier testing.

```suggestion
    if (tags.length === 0) throw new NotFound({ prettyMessage: 'no tags found' })
```